File: CS/SpreadWinDemoCS/shape/freeform.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace SpreadWinDemo.shape
{
    public partial class freeform : SpreadWinDemo.DemoBase
    {
        public freeform()
        {
            InitializeComponent();

            // ワークブックの設定
            InitWorkbook(fpSpread1.AsWorkbook());

            button1.Click += new EventHandler(button1_Click);
            button2.Click += new EventHandler(button2_Click);
        }

        private void InitWorkbook(GrapeCity.Spreadsheet.IWorkbook workbook)
        {
            // フォントの設定
            GrapeCity.Spreadsheet.IStyle normalStyle = workbook.Styles[GrapeCity.Spreadsheet.BuiltInStyle.Normal];
            normalStyle.Font.Name = "メイリオ";
            normalStyle.Font.Size = 9;
        }

        void button1_Click(object sender, EventArgs e)
        {
            fpSpread1.StopAnnotationMode();

            // 図形
            fpSpread1.Features.EnhancedShapeEngine = true;
            fpSpread1.StartAnnotationMode(FarPoint.Win.Spread.AnnotationMode.Freeform);
            fpSpread1.Focus();
        }

        void button2_Click(object sender, EventArgs e)
        {
            fpSpread1.StopAnnotationMode();

            // フリーハンド
            fpSpread1.Features.EnhancedShapeEngine = true;
            fpSpread1.StartAnnotationMode(FarPoint.Win.Spread.AnnotationMode.Scribble);
        }
    }
}

File: VB/SpreadWinDemo/shape/freeform.vb
Public Class freeform

    Public Sub New()

        InitializeComponent()

        ' ワークブックの設定
        InitWorkbook(FpSpread1.AsWorkbook())
    End Sub

    Private Sub InitWorkbook(ByVal workbook As GrapeCity.Spreadsheet.IWorkbook)
        Dim normalStyle As GrapeCity.Spreadsheet.IStyle = workbook.Styles(GrapeCity.Spreadsheet.BuiltInStyle.Normal)
        normalStyle.Font.Name = "メイリオ"
        normalStyle.Font.Size = 9
    End Sub

    Private Sub Button1_Click(sender As Object, e As System.EventArgs) Handles Button1.Click
        FpSpread1.StopAnnotationMode()

        ' 図形
        FpSpread1.Features.EnhancedShapeEngine = True
        FpSpread1.StartAnnotationMode(FarPoint.Win.Spread.AnnotationMode.Freeform)
        FpSpread1.Focus()
    End Sub

    Private Sub Button2_Click(sender As Object, e As System.EventArgs) Handles Button2.Click
        FpSpread1.StopAnnotationMode()

        ' フリーハンド
        FpSpread1.Features.EnhancedShapeEngine = True
        FpSpread1.StartAnnotationMode(FarPoint.Win.Spread.AnnotationMode.Scribble)
    End Sub
End Class
